libbeat/tests/system/test_http.py

```diff
@@ -5,17 +5,22 @@
 
 
 class Test(BaseTest):
+    def setUp(self):
+        super(BaseTest, self).setUp()
+        self.render_config_template()
+        self.proc = self.start_beat(extra_args=["-E", "http.enabled=true"])
+        self.wait_until(lambda: self.log_contains("Starting stats endpoint"))
+
+    def tearDown(self):
+        super(BaseTest, self).tearDown()
+        # Wait till the beat is completely started so it can handle SIGTERM
+        self.wait_until(lambda: self.log_contains("mockbeat start running."))
+        self.proc.check_kill_and_wait()
```

Comment:
Probably good to check that `self.proc` is not `None`, in case setUp failed and `self.proc` was not initialized?

Reply:
Good point. But if `setUp` fails, `tearDown` is not called. So I think this is safe enough for testing code.